pygerm/client/__init__.py

```diff
@@ -11,26 +11,62 @@ def parse_event_payload(data):
     The layout is
 
        "0" [[4 bit chip addr] [5 bit channel addr]] [10 bit TD] [12 bit PD]
-       "1" [31 bit time stamp]
+       "100" [28 bit time stamp]
 
     '''
 
     # TODO sort out if this can be made faster!
+    word1 = data[::2]
+    word2 = data[1::2]
 
     # chip addr
-    chip = (data >> (27)) & 0xf
+    chip = word1 >> 27 & 0xf
     # chan addr
-    chan = (data >> (22)) & 0x1f
+    chan = word1 >> 22 & 0x1f
     # fine ts
-    td = (data >> (12)) & 0x3ff
+    # TODO : why 0x3ff not 1ff??
+    td = word1 >> 12 & 0x1ff
+
     # evergy readings
-    pd = (data) & 0xfff
+    pd = word1 & 0xfff
+
     # FPGA tick
-    ts = data >> 32 & 0x7fffffff
+    # should be 6 f's not 7??
+    ts = word2 & 0x7ffffff
 
     return chip, chan, td, pd, ts
 
 
+def event2payload(chip, chan, td, pd, ts):
+    ''' Convert event data to a payload.
+
+
+    This just creates words of the form:
+
+       "0" [[4 bit chip addr] [5 bit channel addr]] [10 bit TD] [12 bit PD]
+       "100" [28 bit time stamp]
```

Comment:
This does not add up to 32

Reply:
sorry, missed a zero (was copying from the germ daq reference document).
the 1 is correctly shifted by 31 bits though at least in the code. I'll fix now thanks for checking this.